integration-tests/src/integrationTest/org/apache/cassandra/sidecar/acl/RoleBasedAuthorizationIntegrationTest.java

```diff
@@ -463,23 +470,24 @@ void testEndpointRequiringMultipleActions()
                                     "data/multiple_permissions_required_test_keyspace/test_table",
                                     "SNAPSHOT:READ");
 
-            // wait for cache refresh
-            loopAssert(2, 100, () -> {
-                verifyAccess(HttpMethod.GET, listSnapshotRoute, nonAdminClientKeystorePath, response -> {
-                    assertThat(response).isNotNull();
-                    assertThat(response.statusCode()).isEqualTo(HttpResponseStatus.OK.code());
-                    ListSnapshotFilesResponse snapshotFiles = response.bodyAsJson(ListSnapshotFilesResponse.class);
-                    List<ListSnapshotFilesResponse.FileInfo> filesToStream =
-                    snapshotFiles.snapshotFilesInfo()
-                                 .stream()
-                                 .filter(info -> info.fileName.endsWith("-Data.db"))
-                                 .sorted(Comparator.comparing(o -> o.fileName))
-                                 .collect(Collectors.toList());
-                    assertThat(filesToStream).isNotNull().isNotEmpty();
-                    componentDownloadUrl[0] = filesToStream.get(0).componentDownloadUrl();
-                });
+            // Wait for cache to expire without accessing it
+            Uninterruptibles.sleepUninterruptibly(3, TimeUnit.SECONDS);
```

Comment:
nit: 3 seconds is long for unit test. How about explicitly invalidate the cache? I do not think cache eviction is the behavior you want to test in this test case. Invalidating the cache explicitly is deterministic and avoids potential flakiness.